src/com/reco1l/osu/hud/elements/HUDBackButton.kt

```diff
@@ -0,0 +1,133 @@
+package com.reco1l.osu.hud.elements
+
+import com.edlplan.framework.easing.Easing
+import com.reco1l.andengine.*
+import com.reco1l.andengine.shape.Circle
+import com.reco1l.andengine.sprite.ExtendedSprite
+import com.reco1l.framework.ColorARGB
+import com.reco1l.framework.Interpolation
+import com.reco1l.osu.hud.HUDElement
+import org.anddev.andengine.input.touch.TouchEvent
+import ru.nsu.ccfit.zuev.osu.Config
+import ru.nsu.ccfit.zuev.osu.GlobalManager
+import ru.nsu.ccfit.zuev.osu.ResourceManager
+
+class HUDBackButton : HUDElement() {
+
+    private val requiredPressTimeMs = Config.getInt("back_button_press_time", 300).toFloat()
+
+    private val arrow = ExtendedSprite().apply {
+        textureRegion = ResourceManager.getInstance().getTexture("back-arrow")
+        anchor = Anchor.Center
+        origin = Anchor.Center
+
+        relativeSizeAxes = Axes.Both
+        setSize(0.6f, 0.6f)
+    }
+
+    private val backCircle = Circle().apply {
+        setPortion(0f)
+        anchor = Anchor.Center
+        origin = Anchor.Center
+        color = ColorARGB.White
+        depthInfo = DepthInfo.Default
+
+        relativeSizeAxes = Axes.Both
+        setSize(1f, 1f)
+    }
+
+    private val frontCircle = Circle().apply {
+        anchor = Anchor.Center
+        origin = Anchor.Center
+        color = ColorARGB(0xFF002626)
+        depthInfo = DepthInfo.Clear
+
+        relativeSizeAxes = Axes.Both
+        setSize(0.95f, 0.95f)
+    }
+
+
+    private var holdDurationMs = 0f
+        set(value) {
+            field = value.coerceIn(0f, requiredPressTimeMs)
+
+            val progress = if (requiredPressTimeMs > 0) field / requiredPressTimeMs else 0f
+            val scale = 1f + progress / 2f
+
+            alpha = Interpolation.floatAt(holdDurationMs, 0.25f, 0.5f, 0f, requiredPressTimeMs, Easing.OutCubic)
+            backCircle.setPortion(progress)
+            backCircle.setScale(scale)
+            frontCircle.setScale(scale)
+            arrow.setScale(scale)
+        }
+
+
+    private var isPressed = false
+
+
+    init {
+        setSize(SIZE, SIZE)
+        alpha = 0.25f
+
+        attachChild(frontCircle)
+        attachChild(backCircle)
+        attachChild(arrow)
+    }
+
+
+    override fun onManagedUpdate(pSecondsElapsed: Float) {
+
+        if (!isInEditMode) {
+            val realMsElapsed = pSecondsElapsed * 1000
+
+            if (isPressed) {
+                holdDurationMs += realMsElapsed
+
+                if (holdDurationMs >= requiredPressTimeMs) {
+                    isPressed = false
+                    GlobalManager.getInstance().gameScene.pause()
```

Comment:
`holdDurationMs` needs to be reset too after pausing, otherwise the progress will not reset.

Reply:
Yes it does.